client/my-sites/stats/site.jsx

```diff
@@ -168,7 +170,10 @@ class StatsSite extends Component {
 					slug={ slug }
 				/>
 				<div id="my-stats-content">
-					{ config.isEnabled( 'onboarding-checklist' ) && <WpcomChecklist viewMode="banner" /> }
+					{ this.props.isEligibleForChecklist &&
+						! getNeverShowBannerStatus( this.props.siteId ) && (
+							<AsyncLoad require="my-sites/checklist/wpcom-checklist" viewMode="banner" />
```

Comment:
Here's an idea how to implement the async loading in a more reusable way:

*wpcom-checklist/index.js* becomes just a stub that does the async load:
```jsx
const ChecklistLoader = ( { isVisible, ...restProps } ) => isVisible && (
  <AsyncLoad { ...restProps } require="./component.js" />
);
export connect( state => ( { isVisible: isEligible( state ) } ) )( ChecklistLoader );
```

*wpcom-checklist/component.js* now contains the "real" code:
```js
class Checklist extends Component {
  /* load lot of code and render complex UI */
}
```

Now everyone who imports `my-sites/checklist/wpcom-checklist` will get it async-loaded by default, for free. No room for making mistakes like loading it synchronously.

Reply:
Great suggestion! Implemented in the latest version.